Remove the double conditions "options" + "variables" to obtain the di…

rvola opened the pull request:
…scount line with or without taxes

### All Submissions:

-   [X] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [X] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:
This PR fixes identified bug #36794

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes # .

<!-- The next section is mandatory. If your PR doesn't require testing, please indicate that you are purposefully omitting instructions. -->

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Otherwise, please include detailed instructions on how these changes can be tested. Please, make sure to review and follow the guide for writing high-quality testing instructions below. -->

- [x] Have you followed the [Writing high-quality testing instructions guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions)?

1. Depending on whether the user calls the method with 'incl' or 'excl', the result must display the total of allowances with and without taxes

<!-- End testing instructions -->

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

<!-- Mark completed items with an [x] -->

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

vedanshujain commented:
hey @rvola , thanks for this PR. can you also add some unit test for this change please?

rvola committed:
Remove the double conditions "options" + "variables" to obtain the discount line with or without taxes

rvola commented:
I just added a commit for testing

rvola commented:
I need support. I wrote a test.
What should I do ?
thanks

vedanshujain commented:
@rvola , I am planning to take a look this week

rvola commented:
Hello @vedanshujain 
Have you any news

rvola commented:
Hi @vedanshujain 

I just fixed and updated the unit test.
I had a concern in my environment.
Sorry for the push forces, I made it so that the PR only contains useful commits.

Can you validate my work to close?

rvola committed:
Test unit for PR #36885
